File: SkateboardAS/Core/Interfaces/IMachineComponent.cs
using Core.Enums;
using Core.Models;

namespace Core.Interfaces;

public interface IMachineComponent
{
    string Id { get; }
    string Name { get; }
    MachineType Type { get; }
    MachineStatus Status { get; }
    Task<CommandResult> StartAsync(CancellationToken ct = default);
    Task<CommandResult> StopAsync(CancellationToken ct = default);
    Task<CommandResult> ResetAsync(CancellationToken ct = default);
    Task<MachineStatusModel> GetStatusAsync(CancellationToken ct = default);
}

File: SkateboardAS/core/src/main/java/dk/sdu/core/api/IMachineComponent.java
package dk.sdu.core.api;

import dk.sdu.core.enums.MachineStatus;
import dk.sdu.core.enums.MachineType;
import dk.sdu.core.model.CommandResult;
import dk.sdu.core.model.MachineStatusModel;

public interface IMachineComponent {
    String getId();
    String getName();
    MachineType getType();
    MachineStatus getStatus();
    CommandResult start();
    CommandResult stop();
    CommandResult reset();
    MachineStatusModel getStatusDetails();
}
